src/cursor/cur_version.c

```diff
@@ -0,0 +1,387 @@
+             * If the update is a tombstone, we still want to record the stop information but we
+             * also need traverse to the next update to get the full value. If the tombstone was the
+             * last update in the update list, retrieve the ondisk value.
+             */
+            version_cursor->upd_txnid = upd->txnid;
+            version_cursor->upd_durable_stop_ts = upd->durable_ts;
+            version_cursor->upd_stop_ts = upd->start_ts;
+            if (upd->type == WT_UPDATE_TOMBSTONE) {
+                upd = upd->next;
+
+                /* Make sure the next update is not an aborted update. */
+                while (upd != NULL && upd->txnid == WT_TXN_ABORTED)
+                    upd = upd->next;
+            }
+
+            if (upd == NULL)
+                F_SET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED);
+            else {
+                /*
+                 * Set the version cursor's key, which contains all the record metadata for that
+                 * particular version of the update.
+                 */
+                __wt_cursor_set_key(cursor, upd->txnid, upd->durable_ts, upd->start_ts,
+                  version_cursor->upd_txnid, version_cursor->upd_durable_stop_ts,
+                  version_cursor->upd_stop_ts, upd->type, upd->prepare_state, upd->flags,
+                  WT_VERSION_UPDATE_CHAIN);
+
+                /*
+                 * Set the version cursor's value to be the update value. If the update is a modify,
+                 * reconstruct the value.
+                 */
+                if (upd->type != WT_UPDATE_MODIFY)
+                    __wt_upd_value_assign(cbt->upd_value, upd);
+                else
+                    WT_RET(
+                      __wt_modify_reconstruct_from_upd_list(session, cbt, upd, cbt->upd_value));
+
+                __wt_cursor_set_value(cursor, cbt->upd_value->buf);
+
+                upd = upd->next;
+                if (upd == NULL)
+                    F_SET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED);
+                goto done;
+            }
+        }
+    }
+
+    if (!F_ISSET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED)) {
+        WT_RET(__wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &tw));
+
+        if (!WT_TIME_WINDOW_HAS_STOP(&tw)) {
+            ondisk_durable_stop_ts = version_cursor->upd_durable_stop_ts;
+            ondisk_stop_ts = version_cursor->upd_stop_ts;
+            ondisk_stop_txn = version_cursor->upd_txnid;
+        } else {
+            ondisk_durable_stop_ts = tw.durable_stop_ts;
+            ondisk_stop_ts = tw.stop_ts;
+            ondisk_stop_txn = tw.stop_txn;
+        }
+
+        __wt_cursor_set_key(cursor, tw.start_txn, tw.durable_start_ts, tw.start_ts, ondisk_stop_txn,
+          ondisk_durable_stop_ts, ondisk_stop_ts, 0, 0, 0, WT_VERSION_DISK_IMAGE);
+        __wt_cursor_set_value(cursor, cbt->upd_value->buf);
+
+        version_cursor->upd_txnid = tw.start_txn;
```

Comment:
No need to save this any more. hs value has stop anyway.

Reply:
I know that there is a stop_txn field inside the tw structure, but I'm not too sure if any of the hs_cursor->get_key or hs_cursor->get_value will properly get this value.

Reply:
No, you can only get it from the tw structure.